rfcs/0000_conducting_research_while_building_a_stable_platform.md

```diff
@@ -0,0 +1,176 @@
+
+# Drawbacks
+
+### Twice The Work
+
+For any capability that goes through this process, it will be created at least twice. The first time, as a prototype, the second time as a pull request to Monty `main`.
+
+### Implementation Project Is Time Away From Research
+
+Once the prototyped capability is stable and ready for integration with the platform, the researcher is pulled away from research and into software engineering.
+
+### Engineers Become Overwhelmed With Implementation Projects
+
+Building the platform is essential to our vision. Engineers' primary focus is on advancing toward that goal. If there are many implementation projects to integrate, they may take up all available capacity and the progress on the platform itself may slow or come to a halt. Additionally, a bottleneck here could result in the worst of all worlds as we make no progress on the platform and researchers are caught up in software engineering, making no progress on theory.
+
+# Rationale and alternatives
+
+The approach outlined in this RFC is based on the assumption that we will evolve the existing Monty code (`tbp.monty`) into our long-term stable platform. As of this writing, the Monty implementation is closely coupled to research and experimental needs. Implementation projects exist to translate from research and experimental needs to long-term stable platform needs.
+
+### Status Quo
+
+We could maintain the current state of affairs. It is worth noting that the status quo is not the worst possible undesirable feedback loop previously highlighted. What we currently do is a combination of researchers conducting experiments on forks but also merging the experimental code into `tbp.monty`. A multi-track/multi-tasking approach.
+
+```mermaid
+---
+title: Status quo feedback loop
+---
+flowchart LR
+    T(Theory) --> P(Prototype on a fork)
+    P -- Feedback --> T
+    P --> W{Partial functionality?}
+    W -- No --> P
+    W -- Merge --> Monty(Monty main)
+    Monty -- Feedback --> T
+```
+
+With the status quo, the fast feedback loop already exists. Partial functionality is merged when it is deemed ready.
+
+This approach avoids the whole concept of an implementation project, and focuses on integrating the code one pull request at a time. Merging to Monty usually calls for additional design and refactoring. If the merged partial functionality will not change, then merging partial functionality one pull request at a time is the better approach. The challenge comes if the partial functionality changes too often, before the new capability is settled. This leads to multiple rounds of rework.
+
+This RFC proposes to avoid multiple rounds of rework so that researchers can spend what we assume would be most of their time in the prototype-theory feedback loop, followed by batching all the design and refactoring merge work into a single integration project.
+
+### Second System
```

Comment:
I'm assuming this heading is a reference to [second-system effect or second-system syndrome](https://en.wikipedia.org/wiki/Second-system_effect). Would it make sense to change it to one of these to make the connection more explicit (for those familiar with it)?

Reply:
Never mind, I think what you mean here is something like "Build a Second System". You do bring up the second-system effect, but I misread this for some reason.